### ISSUE TITLE
TypeScript: Autocomplete shows more options than a component has

### ISSUE THREAD

**Issue by cintek** [themesberg/flowbite-svelte#1453]
Using v0.46.20 my IDE shows more options for components than they should have. Here an example for `Radio`:

![radio_options](https://github.com/user-attachments/assets/cc497b31-4314-484f-853b-8e0be1724c33)

The cause is that `$$Props` in `src/lib/forms/Radio.svelte` is extending `HTMLInputAttributes`:
`interface $$Props extends HTMLInputAttributes {`

The problem is that a radio input doesn't have such attributes like `max` or `maxLength` and if the IDE makes wrong suggestions you would always have to look into documentation to be sure that a component really has an attribute.

I suggest to define `$$Props` like this (again an example for `Radio`):
```
interface $$Props {
    color?: FormColorType;
    custom?: boolean;
    inline?: boolean;
    group?: number | string | undefined;
    value?: number | string;
    spacing?: string;
    checked?: HTMLInputAttributes["checked"];
  }
```

So, don't extend `HTMLInputAttributes` but map the props to the attributes of `HTMLInputAttributes`. Yes, this is more work for you maintainers of `flowbite-svelte` but it makes using your library easier.

**Comment by shinokada** [comment]
According to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio:

[Additional attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio#additional_attributes)
In addition to the common attributes shared by all [<input>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) elements, radio inputs support the following attributes.

Please note that "the common attributes shared by all <input>". Some user may be using attributes that you may not using. By extending it you can use `required`, `name`, `tabindex`, `accesskey`, etc.

By doing extending it, you won't get any error when you run `pnpm check`'.

So `extends` is intentional.

**Comment by cintek** [comment]
> In addition to the common attributes shared by all elements, radio inputs support the following attributes.

When you take a look at the [attributes of input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attributes) you can find the following sentence:

> Attributes for the <input> element include the [global HTML attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes) and additionally:

Below the sentence you find a table including these rows:

![input1](https://github.com/user-attachments/assets/8e171d8a-c469-4568-b0a9-6f302afe2033)
![input2](https://github.com/user-attachments/assets/aff40121-05f4-41bc-acff-03f35864fde9)

So, from what I understand e.g. a radio input has the [global attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes) and the ones in the table where type is `radio`. In this case `maxLength` doesn't belong to the attributes of a radio input.

**Comment by shinokada** [comment]
That's right. But [svelte](https://github.com/sveltejs/svelte/blob/svelte%405.0.0-next.222/packages/svelte/elements.d.ts) doesn't provide for radio, checkbox etc. They belong to `HTMLInputAttributes`.

**Comment by cintek** [comment]
> That's right. But [svelte](https://github.com/sveltejs/svelte/blob/svelte%405.0.0-next.222/packages/svelte/elements.d.ts) doesn't provide specific types for radio, checkbox etc. They belong to `HTMLInputAttributes`.

Sadly, that's true. I created an issue in the svelte repo: https://github.com/sveltejs/svelte/issues/13447. Maybe they will add some types.

**Comment by FredoMartini** [comment]
Hello,

Another solution would be to use `Omit<>` to remove the extra attributes.

For example for radio :
```ts
interface $$Props extends Omit<HTMLInputAttributes,
    'accept' | 'alt' | 'autocapitalize' | 'autocomplete' | 'capture' |
    'dirname' | 'formaction' | 'formenctype' | 'formmethod' |
    'formnovalidate' | 'formtarget' | height' | 'max' | 'maxlength' |
    'min' | 'minlength' | 'multiple' | 'pattern' | 'placeholder' |
    'popovertarget' | 'popovertargetaction' | 'readonly' | 
    'size' | 'src' | 'step' | 'width'> {
```
